Feature/ot111 35 Endpoint para listar Usuarios

denisdimarco opened the pull request:
[https://alkemy-labs.atlassian.net/browse/OT111-35](https://alkemy-labs.atlassian.net/browse/OT111-35)
-----
-  Created UserController, implemented GET /users
----
- Created UserService, DTOs, Mappers :
> Created UserService & UserServiceImpl, implemented findAllUsers method.
Created RoleDTO & UserDTO
Created AbstractMapper, UserMapper & RoleMapper.
-----
- Fix some inconsistencies :
> -build.gradle: removed duplicated dependency.
-NewsEntity : removed completed TODOs comments
-UserEntity: Added missing @Entity annotation  and @NoArgsConstructor, @Builder, also modified 'Role' name to 'RoleEntity'.
-ActivityRepository: Added missing @Repository annotation.
-RoleRepository: Added missing @Repository annotation.
-UserRepository: Added missing @Repository annotation.

denisdimarco committed:
OT111-35 - Created UserController, implemented GET /users

denisdimarco committed:
OT111-35 - Created UserService, DTOs, Mappers.

Created UserService & UserServiceImpl, implemented findAllUsers method.
Created RoleDTO & UserDTO
Created AbstractMapper, UserMapper & RoleMapper.

denisdimarco committed:
OT111-35 - Fix some inconsistencies

build.gradle: removed duplicated dependency.

NewsEntity : removed completed TODOs comments

UserEntity: Added missing @Entity annotation  and @NoArgsConstructor, @Builder, also modified 'Role' name to 'RoleEntity'.

ActivityRepository: Added missing @Repository annotation.

RoleRepository: Added missing @Repository annotation.

UserRepository: Added missing @Repository annotation.

denisdimarco committed:
Merge branch 'develop' into feature/OT111-35

OsvaldoGWP commented on src/main/java/com/alkemy/ong/model/mapper/UserMapper.java:
```diff
@@ -0,0 +1,39 @@
+                .email(entity.getEmail())
+                .photo(entity.getPhoto())
+                .role(roleMapper.entity2DTO(entity.getRoleId()))
+                .createdDate(entity.getCreatedDate().format(DateTimeFormatter.ISO_OFFSET_DATE_TIME)) //Check if this output format is convenient.
```
por el momento en el dto la fechas  sea el mismo tipo q la entidad, y el formateo q se encargue el front, esto hasta q tengamos una definicion mas clara

OsvaldoGWP commented on src/main/java/com/alkemy/ong/repository/ActivityRepository.java:
```diff
@@ -2,6 +2,8 @@
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
 
+@Repository //IMPORTANT - Added @Repository annotation in branch OT111-35
```
Nota: No hace falta la anotacion ya q eso se va a generar automaticamente cuando se genere la implementación

Igualmente dejo link con una discusión  q tiene mas detalles por si quieres saber

denisdimarco replied:
Me interesa el link!

OsvaldoGWP replied:
https://www.it-swarm-es.com/es/java/repository-no-es-necesario-al-implementar-jparepository/832385271/#:~:text=De%20hecho%2C%20no%20es%20necesario,de%20las%20interfaces%20predefinidas%20Repository%20.&text=Esto%20buscar%C3%A1%20Spring%20Data%20repositories,no%20est%C3%A1n%20anotados%20con%20%40Repository%20.

OsvaldoGWP commented on src/main/java/com/alkemy/ong/model/mapper/AbstractMapper.java:
```diff
@@ -0,0 +1,28 @@
+import java.util.List;
+import java.util.stream.Collectors;
+
+public abstract class AbstractMapper<E, D> { //From sonarsource.com: "This change may not be appropriate in libraries or other applications where the class is intended to be used as an API."
```
este mensaje t esta tirando sonar??

denisdimarco replied:
No, sonar me tira una recomendacion minor, como que convierta la clase abstracta en interface, pero lei que para APIs no aplica...

denisdimarco replied:
Entonces puse un comentario para ver si aplicaria o no...

OsvaldoGWP replied:
ok por el momento lo dejemos asi, despues vamos a hacer un barrido de todo lo que tira sonar

OsvaldoGWP commented on src/main/java/com/alkemy/ong/model/entity/UserEntity.java:
```diff
@@ -36,7 +39,9 @@
     @JoinColumn(name = "role_id")
-    private RoleEntity role;
+
+    private RoleEntity roleId; //IMPORTANT! Changed "Role" name to "RoleEntity" in field class name while working on OT111-35.
```
la variable q se llame role no roleId , porfa

denisdimarco committed:
OT111-35 - PR #19 - Fixed createdDate format in UserDTO

denisdimarco committed:
OT111-35 - Fix PR #19

modified UserMapper method name "getRoleId" => "getRole"
modified UserEntity: variable name "roleId" => role
